File: typescript/src/api/types/ValidationError.ts

export interface ValidationError {
    ctx?: Record<string, unknown> | undefined;
    input?: unknown | undefined;
    loc: ValidationError.Loc.Item[];
    msg: string;
    type: string;
}

export namespace ValidationError {
    export type Loc = Loc.Item[];

    export namespace Loc {
        export type Item = 
            | string
            | number;
    }
}

File: ruby/lib/Sikaru/types/validation_error.rb
# frozen_string_literal: true

module Sikaru
  module Types
    class ValidationError < Internal::Types::Model
      field :ctx, -> { Internal::Types::Hash[String, Object] }, optional: true, nullable: false

      field :input, -> { Object }, optional: true, nullable: false

      field :loc, -> { Internal::Types::Array[Sikaru::Types::ValidationErrorLocItem] }, optional: false, nullable: false

      field :msg, -> { String }, optional: false, nullable: false

      field :type, -> { String }, optional: false, nullable: false
    end
  end
end
